src/Auth0.Core/Exceptions/RateLimitApiException.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Auth0.Core.Http;
+using System;
+
+namespace Auth0.Core.Exceptions
+{
+    /// <summary>
+    /// Represents errors caused by rate limits being exceeded when making API calls.
+    /// </summary>
+    public class RateLimitApiException : ApiException
+    {
+        public RateLimit RateLimit { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class.
+        /// </summary>
+        public RateLimitApiException() {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified error message.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        public RateLimitApiException(string message) : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified error message
+        /// and a reference to the inner exception that is the cause of this exception.
+        /// </summary>
+        /// <param name="message">The error message that explains the reason for the exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception, or a null 
+        /// reference if no inner exception is specified.</param>
+        public RateLimitApiException(string message, Exception innerException)
+            : base(message, innerException)
+        {
+        }
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified <paramref name="rateLimit"/>.
+        /// </summary>
+        /// <param name="rateLimit"><see cref="Http.RateLimit"/> received on the API call that failed.</param>
+        public RateLimitApiException(RateLimit rateLimit)
+            : this("Rate limits exceeded")
```

Comment:
Here's the actual error messaging used from the server:

```
{"statusCode":429,"error":"Too Many Requests","message":"Global limit has been reached","errorCode":"too_many_requests"}
```

Reply:
Hmm, not sure if we want to propagate the server message into an actual exception message, let me think that over.

Reply:
Maybe not propagate directly but maybe use the same text hard-coded here.

Reply:
Will pass actual error in another PR.